xmtp_mls/src/groups/sync.rs

```diff
@@ -612,26 +607,39 @@ impl MlsGroup {
         match intent {
             // Intent with the payload hash matches
             Ok(Some(intent)) => {
+                let intent_id = intent.id;
                 log::info!(
-                    "client [{}] is about to process own envelope [{}]",
+                    "client [{}] is about to process own envelope [{}] for intent [{}]",
                     client.inbox_id(),
-                    envelope.id
-                );
-                log::info!(
-                    "envelope [{}] is equal to intent [{}]",
                     envelope.id,
-                    intent.id
+                    intent_id
                 );
-                self.process_own_message(
-                    client,
-                    intent,
-                    openmls_group,
-                    provider,
-                    message.into(),
-                    envelope.created_ns,
-                    allow_epoch_increment,
-                )
-                .await
+                match self
+                    .process_own_message(
+                        client,
+                        intent,
+                        openmls_group,
+                        provider,
+                        message.into(),
+                        envelope.created_ns,
+                        allow_epoch_increment,
+                    )
+                    .await?
+                {
+                    IntentState::ToPublish => {
+                        Ok(provider.conn_ref().set_group_intent_to_publish(intent_id)?)
+                    }
+                    IntentState::Committed => {
+                        Ok(provider.conn_ref().set_group_intent_committed(intent_id)?)
+                    }
+                    IntentState::Published => {
+                        log::warn!("Unexpected behaviour: returned intent state published from process_own_message");
```

Comment:
For unexpected behavior, I think we should log errors, so that we are more likely to see them. Especially if we choose not to update the state here

Reply:
The problem is that we aren't returning the error in a lot of cases. We only return the error if we want to hard fail without changing the intent status (like a DB connection error). That'll get returned to the calling function, which does log it.

So, we have to log inside `process_own_message` before returning OK

Reply:
What I meant is, as a general rule, we should do `log::error` instead of `log::warn` for cases we expect never to happen - not that we need to return an error to the caller

Reply:
Ahh, I see. I can update that